### PR TITLE
PMM-10679 Fixed expose for mongos on minikube

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-0

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by denisok — APPROVED** [review]

File: service/k8sclient/k8sclient.go [hunk 1]
```diff
@@ -2068,19 +2084,20 @@ func (c *K8sClient) getPSMDBSpec112Plus(params *PSMDBParams, extra extraCRParams
 					MultiAZ: psmdb.MultiAZ{
 						Affinity: extra.affinity,
 					},
-					Expose: extra.expose,
```
[the remark this answers is not in the document]
Comment (gen1us2k) [reply service/k8sclient/k8sclient.go:2071]: We deploy sharded psmdb clusters. In that case, we need to expose only Mongos. For single node clusters we need to expose the primary node but it's not part of the current functionality.

File: service/k8sclient/k8sclient.go [hunk 2]
```diff
@@ -2068,19 +2084,20 @@ func (c *K8sClient) getPSMDBSpec112Plus(params *PSMDBParams, extra extraCRParams
...
 				},
 				Mongos: &psmdb.MongosSpec{
 					Size: params.Size,
 					MultiAZ: psmdb.MultiAZ{
 						Affinity: extra.affinity,
 					},
 					Expose: psmdb.MongosExpose{
-						ExposeType: common.ServiceTypeLoadBalancer,
+						ExposeType: extra.expose.ExposeType,
```
Comment (gen1us2k) [inline service/k8sclient/k8sclient.go:2094]: @BupycHuk Here's the expose logic for PSMDB sharded clusters

Comment (BupycHuk) [reply service/k8sclient/k8sclient.go:2094]: Yeah, but I meant `extra.expose.Enabled`, not `extra.expose.ExposeType`

Comment (gen1us2k) [reply service/k8sclient/k8sclient.go:2094]: [v1.9.0](https://github.com/percona/percona-server-mongodb-operator/releases/tag/v1.9.0)
It was enabled by default in PSMDB 1.9. I think that we can drop it, right?

**Review by percona-csalguero — APPROVED** [review]

File: service/k8sclient/k8sclient.go (hunk 2 above)
Comment (BupycHuk) [reply service/k8sclient/k8sclient.go:2094]: Could you point to the task, please?
As far as I remember users want to be able to not expose their clusters outside of the kubernetes cluster.
If we ignore expose parameters it's a bug.

**Pull request merged**